src/Physics.h

```diff
@@ -1,11 +1,29 @@
 #pragma once
 
+#include "Entity.h"
+#include "OgreApplicationContext.h"
+#include "btBulletDynamicsCommon.h"
+#include "btBulletCollisionCommon.h"
+#include "BulletSoftBody/btSoftRigidDynamicsWorld.h"
+#include "BulletSoftBody/btSoftBodyHelpers.h"
+#include "BulletSoftBody/btSoftBodyRigidBodyCollisionConfiguration.h"
+#include "BulletCollision/CollisionShapes/btShapeHull.h"
+
 struct Physics {
-    const int entityId;
-    bool isImoveable;
-    float weight; // How fast it moves
+    //const int entityId;
+
+    btDefaultCollisionConfiguration* collisionConfiguration;
+    btCollisionDispatcher* dispatcher;
+    btBroadphaseInterface* broadphase;
+    btSequentialImpulseConstraintSolver* solver;
+    btDiscreteDynamicsWorld* dynamicsWorld;
+    std::vector<btRigidBody*> rigidBodies;
+    std::vector<btSoftBody*> softBodies;
 
-    void Update();
+    void update(int timeSinceLastFrame);
```

Comment:
You should probably in GameEngine add a new field for the variable you use here and in the MainGameLoop update that variable every iteration (or just use a pointer).


Then calling Update() should be possible (without the usage of an input parameter)